File: server/api/src/main/java/main/entities/Domicilio/Localidad.java
package main.entities.Domicilio;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@ToString
@Table(name = "localidades", schema = "buen_sabor")
public class Localidad {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "nombre")
    private String nombre;

    @JsonIgnoreProperties(value = {"localidades"}, allowSetters = true)
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_departamento")
    private Departamento departamento;
}

File: src/types/Domicilio/Localidad.ts
import { Departamento } from "./Departamento";
import { Domicilio } from "./Domicilio";

export class Localidad {
    id: number = 0;
    nombre: string = '';
    domicilios: Domicilio[] = [];
    departamento: Departamento | null = null;

    constructor(){

    }
}